pandas/io/excel.py

```diff
@@ -207,21 +209,22 @@ def __init__(self, io, **kwds):
         if engine is not None and engine != 'xlrd':
             raise ValueError("Unknown engine: %s" % engine)
 
-        if isinstance(io, compat.string_types):
-            if _is_s3_url(io):
-                buffer, _, _ = get_filepath_or_buffer(io)
-                self.book = xlrd.open_workbook(file_contents=buffer.read())
-            elif _is_url(io):
-                data = _urlopen(io).read()
-                self.book = xlrd.open_workbook(file_contents=data)
-            else:
-                self.book = xlrd.open_workbook(io)
-        elif engine == 'xlrd' and isinstance(io, xlrd.Book):
+        # If io is a url, want to keep the data as bytes so can't pass
+        # to get_filepath_or_buffer()
+        if _is_url(io):
```

Comment:
If I don't special case URLs here, I get failures in URL related tests,  `get_filepath_or_buffer()` returns the data as `StringIO` and reading from urls fails with errors like: 

```
Traceback (most recent call last):
  File "/home/me/src/pandas/reading_tests.py", line 8, in <module>
    url_table = pd.read_excel(url)
  File "/home/me/src/pandas/pandas/io/excel.py", line 170, in read_excel
    io = ExcelFile(io, engine=engine)
  File "/home/me/src/pandas/pandas/io/excel.py", line 225, in __init__
    self.book = xlrd.open_workbook(file_contents=data)
  File "/home/me/anaconda3/lib/python3.5/site-packages/xlrd/__init__.py", line 441, in open_workbook
    ragged_rows=ragged_rows,
  File "/home/me/anaconda3/lib/python3.5/site-packages/xlrd/book.py", line 91, in open_workbook_xls
    biff_version = bk.getbof(XL_WORKBOOK_GLOBALS)
  File "/home/me/anaconda3/lib/python3.5/site-packages/xlrd/book.py", line 1226, in getbof
    opcode = self.get2bytes()
  File "/home/me/anaconda3/lib/python3.5/site-packages/xlrd/book.py", line 631, in get2bytes
    return (BYTES_ORD(hi) << 8) | BYTES_ORD(lo)
TypeError: unsupported operand type(s) for <<: 'str' and 'int'
```